bookstore/archive.py

```diff
@@ -1,47 +1,89 @@
+"""Archival of notebooks"""
 import json
-import aiobotocore
-
 from asyncio import Lock
 from typing import Dict, NamedTuple
 
+import aiobotocore
 from notebook.services.contents.filemanager import FileContentsManager
 from tornado import ioloop
 
 from .bookstore_config import BookstoreSettings
-
 from .s3_paths import s3_key, s3_display_path
 
 
 class ArchiveRecord(NamedTuple):
+    """Representation of content to archive
+
+    A record to be archived is a tuple containing `filepath`, the `content` to
+    be archived, and the length of time in queue for archiving (`queued_time`).
+    """
     filepath: str
     content: str
     queued_time: float
 
 
 class BookstoreContentsArchiver(FileContentsManager):
+    """Archives notebooks to storage (S3) on save by UI or parameterized workflow
```

Comment:
Can just end this one with `on save`, we don't have any parametrized workflows for this (that I'm aware of, I don't know what this means here).

Reply:
Papermill doesn't have a server or contents manager so if parameterized workflow is referring to that, then this won't be archived.

Reply:
Sure thing. What I was thinking was now or down the road when a notebook is saved (whatever the oirginating source) that it will be  archived. Will end the comment at save.